icu4c/source/common/cmemory.h

```diff
@@ -31,14 +31,25 @@
 #include <stddef.h>
 #include <string.h>
 #include "unicode/localpointer.h"
+#include "uassert.h"
 
 #if U_DEBUG && defined(UPRV_MALLOC_COUNT)
 #include <stdio.h>
 #endif
 
 
-#define uprv_memcpy(dst, src, size) U_STANDARD_CPP_NAMESPACE memcpy(dst, src, size)
-#define uprv_memmove(dst, src, size) U_STANDARD_CPP_NAMESPACE memmove(dst, src, size)
+#define uprv_memcpy(dst, src, size) UPRV_BLOCK_MACRO_BEGIN { \
+    U_ASSERT(dst != NULL); \
```

Comment:
Just as comment/thought: since this header is used by plain .c files, we can't use `nullptr` here unless we wanted to fork the code with something like `#ifdef __cplusplus`. But it's probably cleaner to just use `NULL` here rather than adding an ifdef, but I don't really have a strong opinion on it either way.

Reply:
I am not sure how much C code still uses the internal header files... probably cintltst. And maybe some tools code, although we could (at some point) switch those all to C++ and to using the standard library.

But ok for here & now.